osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/security/providers/HsmEncryptionProvider.java

```diff
@@ -60,10 +60,10 @@ public HsmEncryptionProvider(final File keyStoreFile) {
     public byte[] decrypt(final EncryptedSecret secret, final String keyReference) {
         byte[] decryptedSecret = super.decrypt(secret, keyReference);
         if (decryptedSecret.length > KEY_LENGTH) {
-            final byte[] truncatedDecryptedSecretBytes = Arrays.copyOfRange(decryptedSecret, 0,
-                    decryptedSecret.length-16);
-            LOGGER.trace("Truncating decrypted key from " + Hex.encodeHexString(decryptedSecret) + " to " +
-                            Hex.encodeHexString(truncatedDecryptedSecretBytes));
+            final byte[] truncatedDecryptedSecretBytes = Arrays
+                    .copyOfRange(decryptedSecret, 0, decryptedSecret.length - 16);
```

Comment:
replace 16 by KEY_LENGTH

Reply:
Yeah, I talked to Erik about this some time ago; the way it is programmed here is confusing and only works with particular key sizes and block sizes, which are fortunately always ok. Rewritten this to something more logical. Retest on HSM is required!